app/controllers/admin/words_controller.rb

```diff
@@ -0,0 +1,55 @@
+class Admin::WordsController < ApplicationController
+    before_action :if_not_admin
+  
+    def new
+      @category = Category.find(params[:id])
+      @word = current_category.words.create
+    end
+ 
+    def create
+      @word = current_category.words.create(word_params)
+      if @word.save
+        redirect_to admin_category_words_path
+        flash[:success] = "Created successfully!"
+      else
+        flash[:danger] = "Invalid"
+        render 'new'
+      end  
+    end
+   
+    def edit
+      @category = Category.find(params[:id])
+      @word = @category.words.find(params[:id])
+    end
+ 
+    def update
+      @word = current_category.words.find(params[:id])
+      if @word.update_attributes(word_params)
+        redirect_to admin_category_path
+        flash[:success] = "Saved!"
+      else
+        flash[:danger] = "Invalid"
+        render 'edit'
+      end
+    end
```

Comment:
No need for a current_category. We don't have a fucntion for that.
I recommend using findy_by and adding the ID of the category so that AFTER you are done updating a WORD, you are redirected to the correct category and that category has the WORD you are editing.
![Screen Shot 2021-04-28 at 7 49 01 PM](https://user-images.githubusercontent.com/35232348/116398979-fbcd7080-a85a-11eb-8525-56c339149905.png)

Reply:
Thank you for check. It's still error.  It is not redirected to the correct category and that category has the WORD you are editing.
<img width="1440" alt="スクリーンショット 2021-04-28 21 20 52" src="https://user-images.githubusercontent.com/82156430/116402930-f0347680-a867-11eb-9790-60a526d1e507.png">

Reply:
<img width="1440" alt="スクリーンショット 2021-04-28 21 21 00" src="https://user-images.githubusercontent.com/82156430/116402967-faef0b80-a867-11eb-93e3-ed9da3cd93c1.png">